Add use-case design for login

casker2209 opened the pull request:
The idea is using Security subsystem for encapsulate and Database to manage communication with database

casker2209 committed:
Update use-case-design.rst

The idea is that, giving the Security design mechanism to boundary class and OODB a DBManager whose role is to query. The Security subsystem encapsulates the data into an object  and send it to DBManager to decapsulate it to search for the data

casker2209 committed:
Update source/solution/use-case-design.rst

Co-authored-by: Nguyễn Gia Phong <mcsinyx@disroot.org>

casker2209 committed:
Update source/solution/use-case-design.rst

Co-authored-by: Nguyễn Gia Phong <mcsinyx@disroot.org>

casker2209 committed:
Update source/solution/use-case-design.rst

Co-authored-by: Nguyễn Gia Phong <mcsinyx@disroot.org>

casker2209 committed:
Update use-case-design.rst

casker2209 committed:
Update use-case-design.rst

McSinyx commented on source/solution/use-case-design.rst:
```diff
@@ -10,6 +10,89 @@ Register
+.. uml::
+
+   skinparam defaultFontColor #a80036
+   autonumber "#: //"
```
```suggestion
   autonumber "#:"
```

McSinyx commented on source/solution/use-case-design.rst:
```diff
@@ -10,6 +10,89 @@ Register
+   
+   class AccountData {
+    
+   }
```
Shouldn't this had `verify(authentication information)`?  Again, I'm burnt out, @pdhlong please give a review for the overall model when you have the time.

casker2209 replied:
Nah you have the DBManager for that. I think the normal case is: Query gets result and Query gets no result. If query gets no result the DBManager should handle it like an error ,not the database

McSinyx replied:
What I meant is that AccountData at least know how to react to a message named `verify(authentication information)`.

casker2209 replied:
Change the method name now. Previously I think I should respect the old method from use case analysis. Now since with search() it becomes obsolete so search() will be better

DBManager would handle verify by search itself,so I think AccountData doesn't really need to do anything.

McSinyx replied:
Communication is hard, this is what I really meant:  in sequence diagram, AccountData accept this message

       D -> AccountData: send(account information)

so in the VoPC the method should be listed.

McSinyx replied:
FYI the message name is the method name, i.e. the two diagrams are of *the same* model.

casker2209 replied:
Yeah I just think just using verification is kind of misleading since unlike Register we need the information back,but on second thought it still makes sense. Revert it back now

casker2209 committed:
Update source/solution/use-case-design.rst

Co-authored-by: Nguyễn Gia Phong <mcsinyx@disroot.org>

casker2209 committed:
Update use-case-design.rst

McSinyx commented on source/solution/use-case-design.rst:
```diff
@@ -10,6 +10,89 @@ Register
 Login
 ^^^^^
 
+Interaction Diagrams
+^^^^^^^^^^^^^^^^^^^^
+
+Basic Flow
+""""""""""
```
Notice the heading levels:

```suggestion
Login
^^^^^

Interaction Diagrams
""""""""""""""""""""

Basic Flow
''''''''''
```

McSinyx replied:
Hey you need to fix the ones below as well :wink:

casker2209 committed:
Update source/solution/use-case-design.rst

Co-authored-by: Nguyễn Gia Phong <mcsinyx@disroot.org>

casker2209 committed:
Update use-case-design.rst

Previous use case analysis has verify method,so I thought I should respect that. With the search method though, that would be excessive so I will change the name to reflect it.

casker2209 committed:
Update use-case-design.rst

casker2209 committed:
Update use-case-design.rst

casker2209 committed:
Update use-case-design.rst

casker2209 committed:
Update use-case-design.rst
